src/dependency_injector/resources.py

```diff
@@ -31,7 +18,7 @@ def shutdown(self, resource: T) -> None:
         ...
 
 
-class AsyncResource(Generic[T], metaclass=ResourceMeta):
+class AsyncResource(Generic[T]):
```

Comment:
Does this still not need to inherit from abc.ABC?

Reply:
Hm... It probably needs to. I didn't have a test for unimplemented abc methods. I'll revise this later today. Thanks!

Reply:
Returned ``abc.ABCMeta`` inheritance and added tests in https://github.com/ets-labs/python-dependency-injector/commit/1163ac59d4443c1cfbe2c4391752c83512086b99. I should be more watchful. Thanks for pointing out the issue.